src/colvar_neuralnetworkcompute.cpp

```diff
@@ -146,13 +161,15 @@ void denseLayer::readFromFile(const std::string& weights_file, const std::string
             m_weights.push_back(weights_tmp);
         }
     }
+    proxy->close_input_stream(weights_file);
+
     // parse biases file
-    std::ifstream ifs_biases(biases_file.c_str());
-    if (!ifs_biases) {
-        throw std::runtime_error("Cannot open file " + biases_file);
+    std::istream *ifs_biases = proxy->input_stream(biases_file, "biases file");
+    if (ifs_biases == NULL) {
+        return;
```

Comment:
My understanding is that here the return value is a pointer solely to be able to track errors through null pointers. Instead of a pointer, could input_stream() return a reference to an istream that is marked as bad (`stream.clear(ios::badbit)`) in case of error? This would preserve more of the syntax on the caller side.

Reply:
Ok, but that means that the entire framework always needs to rely on `ifstream` after all... You can't explicitly allocate an `istream` object.

Reply:
I think `istream` also supports being converted to `bool` for checking these bits (https://en.cppreference.com/w/cpp/io/basic_ios/operator_bool).

Reply:
@jhenin You came back with this request three times, which means that you feel very strongly about this design choice. I took a stab at it for a bit, but stumbled with two additional complications.
1. Returning a reference to an invalid object means that I could not initialize it right in the return statement, and I had to allocate a specific object just to return an invalid one. This object is return to threads > 0 if there is a bug, which I hope means that the code crashes before multiple threads crash the memory. In the back of my mind, I'm still concerned about deadlocks.
2. References cannot be reassigned, so those cases where you are reusing the same stream object become more complex, not simpler.

I will be busy for several days, so I put the work in progress in this branch and putting it back on draft status. At least the memory leaks should be addressed.